### ISSUE TITLE
RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded

### ISSUE THREAD

**Issue by mileslane** [kellyselden/git-diff-apply#283]
Running:
$ yarn ember-cli-update --to 3.10.0
I got:
{ RangeError [ERR_CHILD_PROCESS_STDIO_MAXBUFFER]: stdout maxBuffer length exceeded
    at Socket.onChildStdout (child_process.js:354:14)
    at Socket.emit (events.js:198:13)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:265:13)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:94:17)
  cmd: 'git commit -m "message" --no-verify',

ember-cli-update is using git-diff-apply@0.19.4.
ember-cli-update is 0.43.3.

To work around this problem, I made this edit in git-diff-apply/src/git-remove-all.js:

```
*** 9,14 ****
--- 9,15 ----
    // of output if it gets run in a large repo.
    // See https://github.com/kellyselden/git-diff-apply/pull/277
    let files = (await runWithSpawn('git', ['ls-files'], options))
+     .trim()
      .split(/\r?\n/g)
      .filter(Boolean);
```

**Comment by mileslane** [comment]
I should mention, I am not a developer and it occurs to me that while the change I made got the program to run and package.json got updated, perhaps trimming was the wrong thing to do.

**Comment by kellyselden** [comment]
Hmm I don't see how that change would fix anything. The following `.filter(Boolean)` should make that trim unnecessary.

**Comment by timiyay** [comment]
This appears to be a similar issue to https://github.com/ember-cli/ember-cli-update/issues/583.

A PR to fix that issue may also fix this one: https://github.com/kellyselden/git-diff-apply/pull/295

**Comment by kellyselden** [comment]
Should be fixed as of 0.22.0.
